app/views/chats/_message.html.erb

```diff
@@ -0,0 +1,41 @@
+<div class="mb-4 flex items-start gap-3" id="message_<%= message.id %>">
+  <div class="flex-shrink-0">
+    <% if message.role == 'user' %>
+      <div class="w-8 h-8">
+        <%= render "settings/user_avatar", user: message.user %>
+      </div>
+    <% else %>
+      <div class="w-8 h-8">
+        <%= render "shared/circle_logo", name: "AI", hex: "#22CCEE", size: "full" %>
+      </div>
+    <% end %>
+  </div>
+  <div class="flex-1">
+    <div class="inline-block max-w-full">
+      <div class="text-sm prose text-gray-900" id="message_content_<%= message.id %>">
+        <% if message.content == '...' %>
```

Comment:
You could move some of the logic to the model so the magic `'...'` string logic isn't in several different places. Could be something like `message.pending_message?` or something like that 🤷🏻.